Directory.Build.props

```diff
@@ -22,22 +22,22 @@
   </PropertyGroup>
 
   <!--
-    Arcade only allows the revision to contain up to two characters, and AppVeyor does not roll-over
+    Arcade only allows the revision to contain up to two characters, and GitHub Actions does not roll-over
     build numbers every day like Azure DevOps does. To balance these two requirements, set the official
     build ID to be the same format as the built-in default from Arcade, except with the revision number
     being the number of the quarter hour of the current time of day (24 * 4 = 96, which is less than 100).
     So a build between 00:00 and 00:14 would have a revision of 1, and a build between 23:45 and 23:59:59
     would have a revision of 97.
   -->
 
-  <PropertyGroup Condition=" '$(APPVEYOR)' == 'True' AND '$(APPVEYOR_PULL_REQUEST_NUMBER)' == '' ">
+  <PropertyGroup Condition=" '$(GITHUB_ACTIONS)' == 'true' AND '$(GITHUB_REF.StartsWith(`refs/pull/`))' == 'false' ">
```

Comment:
FYI: while it's a great workaround, it sadly proved to be a bit unreliable in the OpenIddict repo, where builds tend to last longer than in this repo: let's say you push a commit at 14:25, if `dotnet build/pack` is still running at 14:30, then the packages built and packed after 14:30 will get a different version number than the previous ones, which causes issues when the packages produced after depend on packages built before. It's very likely because these properties are always re-evaluated per project.

Now that we're moving to GitHub actions, do you think we could move this logic to a custom action?

Reply:
Ah interesting. So you mean an action that would run at the start of the build and set the number in an environment variable, and then the build would just reference that so it's constant for that specific build?

Reply:
Yes, absolutely! Or maybe there's a way to compute that directly in the build.yml file using inline expressions? 🤷‍♂️

Reply:
I know you can persist env vars - I'd be happy to look into it. OK for me to do that as a follow up PR?

Reply:
> OK for me to do that as a follow up PR?

Sure, thanks! 👏

Reply:
#448